src/js/libgeo/GeoRender.js

```diff
@@ -99,6 +104,27 @@ function drawgeoline(el) {
         if (zz.value.real >= 0) { // finite segment
             evaluator.draw$2(
                 [el.startpos, el.endpos], modifs);
+            if (el.labeled && !el.tmp) {
+                var lbl = el.printname || el.name || "S";
+                var l = List.cross(el.startpos, el.endpos);
```

Comment:
I think the information is already given in ``el.homog``.

Reply:
We need the oriented line here (because the label is supposed to be on the same always side). Due to the normalization, the orientation sometimes got lost using `el.homog` causing jumps of the label. But there was a small bug, because also the points were not oriented as well.